pypeit/core/flexure.py

```diff
@@ -303,147 +305,226 @@ def spec_flex_shift(obj_skyspec, arx_skyspec, arx_lines, mxshft=20):
                 corr_cen=corr.size/2, smooth=smooth_sig_pix, success=success)
 
 
-def spec_flexure_obj(specobjs, slitord, bpm, method, sky_file, mxshft=None):
-    """Correct wavelengths for flexure, object by object
+def spec_flexure_slit(slits, slitord, slitmask, bpm, sky_file, method="boxcar", specobjs=None, slitspecs=None,
+                      waveimg=None, mxshft=None, slitShift=None):
+    """Correct wavelengths for flexure, slit by slit
 
     Args:
-        specobjs (:class:`pypeit.specobjs.Specobjs`):
+        slits (:class:`pypeit.slittrace.SlitTraceSet`_):
+            Slit trace set
         slitord (`numpy.ndarray`_):
             Array of slit/order numbers
+        slitmask (`numpy.ndarray`_):
+            Slit mask
         bpm (`numpy.ndarray`_):
             True = masked slit
-        method (:obj:`str`)
-          'boxcar' -- Recommneded
-          'slitpix' --
+        method (:obj:`str`, optional)
+          'boxcar' -- Recommended for object extractions
+          'slitcen' -- Recommended when no objects are being extracted
         sky_file (str):
             Sky file
+        specobjs (:class:`pypeit.specobjs.Specobjs`_, optional):
+            Spectral extractions
+        slitspecs (list, optional):
```

Comment:
line tools dependencies should not be in this routine. This may be left over from the past, but we agreed to get rid of these.

Reply:
Also add to the documentation that this slit specs contain sky spectra extracted for each slit.